lib/versions.ps1

```diff
@@ -1,53 +1,175 @@
+    $DifferenceVersion = $DifferenceVersion -replace '\+', '-'
 
-        if($ver_a[$i] -gt $ver_b[$i]) { return 1; }
-        if($ver_a[$i] -lt $ver_b[$i]) { return -1; }
+    if ($DifferenceVersion -eq $ReferenceVersion) {
+        return 0
     }
-    if($ver_b.length -gt $ver_a.length) { return -1 }
-    return 0
-}
 
-function qsort($ary, $fn) {
-    if($null -eq $ary) { return @() }
-    if(!($ary -is [array])) { return @($ary) }
+    $splitReferenceVersion = @($ReferenceVersion -split $Delimiter | ForEach-Object { if ($_ -match "^\d+$") { [Long]$_ } else { ($_ -replace '[a-zA-Z]+', '.$&.').Replace('..', '.').Trim('.') } })
+    $splitDifferenceVersion = @($DifferenceVersion -split $Delimiter | ForEach-Object { if ($_ -match "^\d+$") { [Long]$_ } else { ($_ -replace '[a-zA-Z]+', '.$&.').Replace('..', '.').Trim('.') } })
 
-    $pivot = $ary[0]
-    $rem = $ary[1..($ary.length-1)]
+    if ($splitReferenceVersion[0] -eq 'nightly' -and $splitDifferenceVersion[0] -eq 'nightly') {
+        return 0
+    }
 
-    $lesser = qsort ($rem | Where-Object { (& $fn $_ $pivot) -lt 0 }) $fn
+    for ($i = 0; $i -lt [Math]::Max($splitReferenceVersion.Length, $splitDifferenceVersion.Length); $i++) {
+        if ($i -ge $splitReferenceVersion.Length) {
+            if ($splitDifferenceVersion[$i] -match "alpha|beta|rc|pre") {
+                return -1
+            } else {
+                return 1
+            }
+        }
+        if ($i -ge $splitDifferenceVersion.Length) {
+            if ($splitReferenceVersion[$i] -match "alpha|beta|rc|pre") {
+                return 1
+            } else {
+                return -1
+            }
+        }
 
-    $greater = qsort ($rem | Where-Object { (& $fn $_ $pivot) -ge 0 }) $fn
+        if (($splitReferenceVersion[$i] -match "\.") -or ($splitDifferenceVersion[$i] -match "\.")) {
+            $Result = Compare-Version $splitReferenceVersion[$i] $splitDifferenceVersion[$i] -Delimiter '\.'
+            if ($Result -ne 0) {
+                return $Result
+            } else {
+                continue
+            }
+        }
+
+        if ($null -ne $splitReferenceVersion[$i] -and $null -ne $splitDifferenceVersion[$i]) {
+            # don't try to compare int to string
+            if ($splitReferenceVersion[$i] -is [string] -and $splitDifferenceVersion[$i] -isnot [string]) {
+                $splitDifferenceVersion[$i] = "$($splitDifferenceVersion[$i])"
+            }
+            if ($splitDifferenceVersion[$i] -is [string] -and $splitReferenceVersion[$i] -isnot [string]) {
+                $splitReferenceVersion[$i] = "$($splitReferenceVersion[$i])"
+            }
+        }
 
-    return @() + $lesser + @($pivot) + $greater
+        if ($splitDifferenceVersion[$i] -gt $splitReferenceVersion[$i]) { return 1 }
+        if ($splitDifferenceVersion[$i] -lt $splitReferenceVersion[$i]) { return -1 }
+    }
+
+    return 0
 }
-function sort_versions($versions) { qsort $versions compare_versions }
+
+# Deprecated
```

Comment:
Do not just comment out these functions. Yes it is pure core function, but maybe someone could abuse it inside manifests or other tools, which are using scoop modules. 

Add comment above the function and warn into function body with deprication message.